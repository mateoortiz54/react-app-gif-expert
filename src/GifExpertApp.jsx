import { useState } from "react";
import { AddCategory, GifGrid } from "./components";
// import { GifGrid } from "./components/GifGrid";


export const GifExpertApp = () => {


    const [categories, setCategories] = useState(['One Punch'])



    const onAddCategory = (newCategory) =>{
        
        if (categories.includes(newCategory)) return;

        
        // Una forma de hacerlo
        setCategories([newCategory, ...categories]);
        // setCategories( cat => [...cat, 'Tokyo Ghoul'])
    };

    return (
        <>
            {/* titulo */}
            <h1>GifExpertApp</h1>

            {/* input para buscar el gif */}
            <AddCategory onNewCategory={event => onAddCategory(event)}/>

            {/* listado de Gifs */}
            {/* <button onClick={onAddCategory}>Add Category</button> */}

            <ol>
                {categories.map(category => (
                    <GifGrid  category={category} key={category}/>
                ))}
            </ol>
                {/* Gif Item */}


        </>
        
    )
}

