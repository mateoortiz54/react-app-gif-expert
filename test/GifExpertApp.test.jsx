import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"
// jest.mock('react')

describe('Pruebas ene le componente GifExpertApp', () => {
    test('debe de mostrar el titulo del componente', () => {

        render(<GifExpertApp/>)
        screen.debug();
        expect(screen.getByText('GifExpertApp')).toBeTruthy();

            
    })
    test('debe de concordar con el snapshoot', () => {
        const {container} = render(<GifExpertApp/>)
        expect(container).toMatchSnapshot();
    })
})