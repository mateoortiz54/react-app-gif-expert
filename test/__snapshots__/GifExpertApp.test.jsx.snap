// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas ene le componente GifExpertApp debe de concordar con el snapshoot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <div>
      <h3>
        One Punch
      </h3>
      <h2>
        Cargando imagenes....
      </h2>
      <div
        class="card-grid"
      />
    </div>
  </ol>
</div>
`;
